packages/playwright-core/src/utils/happy-eyeballs.ts

```diff
@@ -66,7 +67,41 @@ export async function createSocket(host: string, port: number): Promise<net.Sock
   });
 }
 
-async function createConnectionAsync(options: http.ClientRequestArgs, oncreate: ((err: Error | null, socket?: net.Socket) => void) | undefined, useTLS: boolean) {
+export async function createTLSSocket(options: tls.ConnectionOptions): Promise<tls.TLSSocket> {
+  return new Promise((resolve, reject) => {
+    assert(options.host, 'host is required');
+    if (net.isIP(options.host)) {
+      const socket = tls.connect(options)
+      socket.on('connect', () => resolve(socket));
+      socket.on('error', error => reject(error));
+    } else {
+      createConnectionAsync(options, (err, socket) => {
```

Comment:
I wonder if `createConnectionAsync` should allow IP addresses and deal with them? This would remove the ip-specific logic from all other callers of it. Maybe in a follow up if you like that.